packages/@aws-cdk/aws-appsync/test/integ.graphql-s3.ts

```diff
@@ -0,0 +1,61 @@
+import * as path from 'path';
+import * as db from '@aws-cdk/aws-dynamodb';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as s3deploy from '@aws-cdk/aws-s3-deployment';
+import * as cdk from '@aws-cdk/core';
+import * as appsync from '../lib';
+
+/*
+ * Creates an Appsync GraphQL API with schema definition through
+ * s3 location.
+ *
+ * Stack verification steps:
+ * Deploy app and check if schema is defined.
+ *
+ * -- cdk deploy --app 'node integ.graphql-s3.js'             -- start                      --
+ * -- aws appsync list-graphql-apis                           -- obtain api id && endpoint  --
+ * -- aws appsync list-api-keys --api-id [API ID]             -- obtain api key             --
+ * -- bash verify.integ.graphql-s3.sh [APIKEY] [ENDPOINT]     -- return with "getTests"     --
+ * -- cdk destroy --app 'node integ.graphql-s3.js'            -- clean                      --
+ */
+
+const app = new cdk.App();
+
+const stack = new cdk.Stack(app, 'aws-appsync-integ');
+
+const bucket = new s3.Bucket(stack, 'Bucket');
+
+const deploy = new s3deploy.BucketDeployment(stack, 'Deployment', {
+  sources: [s3deploy.Source.asset(path.join(__dirname, 'schemas'))],
+  destinationBucket: bucket,
+});
+
+const api = new appsync.GraphqlApi(stack, 'Api', {
+  name: 'integ-test-s3',
+  schema: appsync.Schema.fromS3Location({
+    bucket,
+    key: 'appsync.test.graphql',
+  }),
+});
+
+api.node.addDependency(deploy);
```

Comment:
Really don't like the fact that you have to call this yourself.

Should we have a `Schema` variant that takes in the S3 deployment type, that handles this for you?

Reply:
Hm.. i think this is residual code that I had before when I was trying to get the bucket token to resolve

Reply:
Actually nevermind, I retract that previous statement. Without the dependency between the deployment and schema, CDK doesn't know to that it needs to deploy the Schema after running the deployment. 

We currently have an API for the api called `addSchemaDependency` for the `GraphqlApi` construct.

If we want to make this a smoother process we could do the following:

```ts
export class S3Schema extends SchemaImpl {
  ...
  public constructor(bucket: IBucket, key: string, deployment?: BucketDeployment) {
     ...
     this.deployment = deployment.
  }

  public bind(api) {
    ...
    this.schema.node.addDependency(this.deployment);
  }
```

Reply:
This needs to be a separate static factory option that takes in a `BucketDeployment` (not both an `IBucket` and a `BucketDeployment`, because `BucketDeployment` already has an `IBucket` defined).

Reply:
ah yes that makes sense

Reply:
Should that change occur in another PR? And keep this one focused on just `fromS3Location`? In fact, I wouldn't mind changing the static function name to `fromBucket`

Reply:
Sure.